libs/wire-api/src/Wire/API/Call/Config.hs

```diff
@@ -93,36 +99,71 @@ import Wire.API.Arbitrary (Arbitrary (arbitrary), GenericUniform (..))
 -- | A configuration object resembling \"RTCConfiguration\"
 --
 -- The \"ttl\" field is a proprietary extension
+-- The \"sft_servers\" field is a proprietary extension
 --
 -- cf. https://developer.mozilla.org/en-US/docs/Web/API/RTCPeerConnection/RTCPeerConnection#RTCConfiguration_dictionary
 data RTCConfiguration = RTCConfiguration
-  { _rtcConfIceServers :: List1 RTCIceServer,
+  { _rtcConfIceServers :: NonEmpty RTCIceServer,
+    _rtcConfSftServers :: Maybe (NonEmpty SFTServer),
     _rtcConfTTL :: Word32
   }
   deriving stock (Eq, Show, Generic)
   deriving (Arbitrary) via (GenericUniform RTCConfiguration)
 
-rtcConfiguration :: List1 RTCIceServer -> Word32 -> RTCConfiguration
+rtcConfiguration :: NonEmpty RTCIceServer -> Maybe (NonEmpty SFTServer) -> Word32 -> RTCConfiguration
 rtcConfiguration = RTCConfiguration
 
 modelRtcConfiguration :: Doc.Model
 modelRtcConfiguration = Doc.defineModel "RTCConfiguration" $ do
   Doc.description "A subset of the WebRTC 'RTCConfiguration' dictionary"
   Doc.property "ice_servers" (Doc.array (Doc.ref modelRtcIceServer)) $
     Doc.description "Array of 'RTCIceServer' objects"
+  Doc.property "sft_servers" (Doc.array (Doc.ref modelRtcSftServer)) $
+    Doc.description "Array of 'SFTServer' objects (optional)"
   Doc.property "ttl" Doc.int32' $
     Doc.description "Number of seconds after which the configuration should be refreshed (advisory)"
 
 instance ToJSON RTCConfiguration where
-  toJSON (RTCConfiguration srvs ttl) =
+  toJSON (RTCConfiguration srvs sfts ttl) =
     object
       [ "ice_servers" .= srvs,
+        "sft_servers" .= sfts,
         "ttl" .= ttl
       ]
 
 instance FromJSON RTCConfiguration where
   parseJSON = withObject "RTCConfiguration" $ \o ->
-    RTCConfiguration <$> o .: "ice_servers" <*> o .: "ttl"
+    RTCConfiguration <$> o .: "ice_servers" <*> o .: "sft_servers" <*> o .: "ttl"
+
+--------------------------------------------------------------------------------
+-- SFTServer
+
+newtype SFTServer = SFTServer
+  { _sftURL :: HttpsUrl
+  }
+  deriving stock (Eq, Show, Ord, Generic)
+  deriving (Arbitrary) via (GenericUniform SFTServer) --TODO is this correct?
+
+instance ToJSON SFTServer where
+  toJSON (SFTServer url) =
+    object
+      [ "urls" .= [url]
+      ]
+
+instance FromJSON SFTServer where
+  parseJSON = withObject "SFTServer" $ \o ->
+    o .: "urls" >>= \case
+      [url] -> pure $ SFTServer url
+      xs -> fail $ "SFTServer can only have exactly one URL, found " <> show (length xs)
+
+sftServer :: HttpsUrl -> SFTServer
+sftServer = SFTServer
+
+modelRtcSftServer :: Doc.Model
+modelRtcSftServer = Doc.defineModel "RTC SFT Server" $ do
+  Doc.description "Inspired by WebRTC 'RTCIceServer' object, contains details of SFT servers"
+  Doc.property "urls" (Doc.array Doc.string') $
+    Doc.description "Array of SFT server addresses of the form 'https://<addr>:<port>'"
```

Comment:
Can we assume port 443, or will we be explicit and specify 443 in the server URLs?

Reply:
Stripping the 443 is a bit of work and having it doesn't really cause any issues, so I would say it is fine if we're always explicit about the port.